cli/internal/lockfile/berry_lockfile.go

```diff
@@ -1,39 +1,78 @@
 package lockfile
 
 import (
-	"bufio"
 	"bytes"
+	"encoding/json"
 	"fmt"
 	"io"
+	"regexp"
+	"sort"
+	"strconv"
 	"strings"
 
+	"github.com/Masterminds/semver"
 	"github.com/pkg/errors"
 	"github.com/vercel/turborepo/cli/internal/turbopath"
 	"gopkg.in/yaml.v3"
 )
 
-// YarnLockfileEntry package information from yarn lockfile
-type YarnLockfileEntry struct {
-	// resolved version for the particular entry based on the provided semver revision
-	Version   string `yaml:"version"`
-	Resolved  string `yaml:"resolved"`
-	Integrity string `yaml:"integrity"`
-	// the list of unresolved modules and revisions (e.g. type-detect : ^4.0.0)
-	Dependencies map[string]string `yaml:"dependencies,omitempty"`
-	// the list of unresolved modules and revisions (e.g. type-detect : ^4.0.0)
-	OptionalDependencies map[string]string `yaml:"optionalDependencies,omitempty"`
+var _multipleKeyRegex = regexp.MustCompile(" *, *")
+
+// A tag cannot start with a "v"
+var _tagRegex = regexp.MustCompile("^[a-zA-Z0-9.-_-[v]][a-zA-Z0-9._-]*$")
+
+var _metadataKey = "__metadata"
+
+type _void struct{}
+
+// BerryLockfileEntry package information from yarn lockfile
+// Full Definition at https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-core/sources/Manifest.ts
+// Only a subset of full definition are written to the lockfile
+type BerryLockfileEntry struct {
+	Version      string `yaml:"version"`
+	LanguageName string `yaml:"languageName,omitempty"`
+
+	Dependencies     map[string]string `yaml:"dependencies,omitempty"`
+	PeerDependencies map[string]string `yaml:"peerDependencies,omitempty"`
+
+	DependenciesMeta     map[string]BerryDependencyMetaEntry `yaml:"dependenciesMeta,omitempty"`
+	PeerDependenciesMeta map[string]BerryDependencyMetaEntry `yaml:"peerDependenciesMeta,omitempty"`
+
+	Bin map[string]string `yaml:"bin,omitempty"`
+
+	LinkType   string `yaml:"linkType,omitempty"`
+	Resolution string `yaml:"resolution,omitempty"`
+	Checksum   string `yaml:"checksum,omitempty"`
+	Conditions string `yaml:"conditions,omitempty"`
```

Comment:
Hi! If you're using the same YAML parser that existed before, you may need to order these based on their presence in the lockfile definition.

This order follows the order present in the Yarn Berry lockfile definition:

```go
type BerryLockfileEntry struct {
 	// resolved version for the particular entry based on the provided semver revision
 	Version    string `yaml:"version"`
 	CacheKey   string `yaml:"cacheKey,omitempty"` // only present in the __metadata object
 	Resolution string `yaml:"resolution,omitempty"`
 	// the list of unresolved modules and revisions (e.g. type-detect : ^4.0.0)
 	Dependencies map[string]string `yaml:"dependencies,omitempty"`
 	// the list of unresolved modules and revisions (e.g. type-detect : ^4.0.0)
 	PeerDependencies     map[string]string          `yaml:"peerDependencies,omitempty"`
 	PeerDependenciesMeta map[string]map[string]bool `yaml:"peerDependenciesMeta,omitempty"`
 	Bin                  map[string]string          `yaml:"bin,omitempty"`
 	OptionalDependencies map[string]string          `yaml:"optionalDependencies,omitempty"`
 	Checksum             string                     `yaml:"checksum,omitempty"`
 	Conditions           string                     `yaml:"conditions,omitempty"`
 	LanguageName         string                     `yaml:"languageName,omitempty"`
 	LinkType             string                     `yaml:"linkType,omitempty"`
 }
```

My version is unintentionally missing `LanguageName`, and you may not need `CacheKey` as that's just part of the metadata.

If the order doesn't matter due to the implementation approach, you can ignore this if you'd like!

Reply:
I'm just using `yaml` to parse the lockfile. Sadly due to some oddities with the Yarn lockfile format I had to [port the serialization logic](https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-parsers/sources/syml.ts) to Go instead of using the `yaml` library.